appservice/src/deploy/runWithZipStream.ts

```diff
@@ -71,14 +71,26 @@ const commonGlobSettings: {} = {
 /**
  * Adds files using glob filtering
  */
-async function getFilesFromGlob(folderPath: string): Promise<string[]> {
+async function getFilesFromGlob(folderPath: string, client: SiteClient): Promise<string[]> {
     const zipDeployConfig: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration(ext.prefix, vscode.Uri.file(folderPath));
     const globOptions = { cwd: folderPath, followSymbolicLinks: true, dot: true };
     const globPattern: string = zipDeployConfig.get<string>('zipGlobPattern') || '**/*';
     const filesToInclude: string[] = await globby(globPattern, globOptions);
+    const zipIgnorePatternStr = 'zipIgnorePattern';
 
-    const ignorePattern: string | string[] = zipDeployConfig.get<string | string[]>('zipIgnorePattern') || '';
-    const filesToIgnore: string[] = await globby(ignorePattern, globOptions);
+    let ignorePatternList: string | string[] = zipDeployConfig.get<string | string[]>(zipIgnorePatternStr) || '';
+    const filesToIgnore: string[] = await globby(ignorePatternList, globOptions);
+
+    if (ignorePatternList) {
+        if (typeof ignorePatternList === 'string') {
+            ignorePatternList = [ignorePatternList];
+        }
+        ext.outputChannel.appendLog(localize('zipIgnoreFileMsg', `Ignoring files from \"{0}.{1}\"`, ext.prefix, zipIgnorePatternStr), { resourceName: client.fullName });
+        ext.outputChannel.appendLine(`\"${ignorePatternList}\"`);
```

Comment:
Mmm I think you just added this line for debugging purposes?

Reply:
No, this part of the expected log output.

Reply:
Why? It seems duplicate with the for loop going through `ignorePatternList` again just below it

Reply:
Yes, you are right. thank you!